Test/CoreSDK.Test/Net40/Extensibility/Implementation/TelemetryConfigurationFactoryTest.cs

```diff
@@ -350,6 +350,137 @@ public void LoadInstanceCreatesNewInstanceOfExpectedTypeWhenPropertiesAreSpecifi
+            try
+            {
+                var configuration = new TelemetryConfiguration();
+                new TestableTelemetryConfigurationFactory().Initialize(configuration);
+
+                // Assume that LoadFromXml method is called, tested separately
+                Assert.True(configuration.TelemetryProcessor !=null);
+                Assert.IsType<StubTelemetryProcessor>(configuration.TelemetryProcessor);
+
+                //validate the chain linking stub1->stub2->transmission
+                var tp1 = (StubTelemetryProcessor) configuration.TelemetryProcessor;
+                var tp2 = (StubTelemetryProcessor2) tp1.next;
+                var tpLast = (TransmissionProcessor) tp2.next;
+
+            }
+            finally
+            {
+                PlatformSingleton.Current = null;
+            }
+        }
+
+        [TestMethod]
+        public void InitializeTelemetryProcessorFromConfigurationFile()
+        {
+            string configFileContents = Configuration(
+                @"
+                  <TelemetryInitializers>
+                  <Add Type=""Microsoft.ApplicationInsights.TestFramework.StubTelemetryInitializer, Microsoft.ApplicationInsights.TestFramework"" />
+                  </TelemetryInitializers>
+                  <TelemetryProcessors>
+                  <Add Type=""Microsoft.ApplicationInsights.TestFramework.StubTelemetryProcessor, Microsoft.ApplicationInsights.TestFramework"" />                  
+                  </TelemetryProcessors>"
+                );
+            var platform = new StubPlatform { OnReadConfigurationXml = () => configFileContents };
+            PlatformSingleton.Current = platform;
+            try
+            {
+                var configuration = new TelemetryConfiguration();
+                new TestableTelemetryConfigurationFactory().Initialize(configuration);
+
+                // Assume that LoadFromXml method is called, tested separately
+                Assert.True(configuration.TelemetryProcessor != null);
+                Assert.IsType<StubTelemetryProcessor>(configuration.TelemetryProcessor);
+
+                //validate the chain linking stub1->transmission
+                var stub1 = (StubTelemetryProcessor)configuration.TelemetryProcessor;
+                var transmission = (TransmissionProcessor)stub1.next;                
+            }
+            finally
+            {
+                PlatformSingleton.Current = null;
+            }
+        }
+
+        [TestMethod]
+        public void InitializeTelemetryProcessorFromConfigurationFileWhenNoTelemetryProcessorsTagSpecified()
+        {
+            // no TelemetryProcessors - TransmissionProcessor should be automatically created.
+            string configFileContents = Configuration(
+                @"
+                  <TelemetryInitializers>
+                  <Add Type=""Microsoft.ApplicationInsights.TestFramework.StubTelemetryInitializer, Microsoft.ApplicationInsights.TestFramework"" />
```

Comment:
do not take dependency on test framework here. You can define telemetry initializer here if needed.